mlflow/deployments/openai/__init__.py

```diff
@@ -0,0 +1,279 @@
+                # Invalid inputs:
+                #   - Wrong engine + correct/wrong deployment_id
+                #   - No engine + wrong deployment_id
+                # Valid inputs:
+                #   - Correct engine + correct/wrong deployment_id
+                #   - No engine + correct deployment_id
+                if deployment_id is not None:
+                    _logger.warning(
+                        "Both engine and deployment_id are set. "
+                        "Using engine as it takes precedence."
+                    )
+                inputs = {"engine": engine, **inputs}
+            elif deployment_id is None:
+                raise MlflowException(
+                    "Either engine or deployment_id must be set for Azure OpenAI API",
+                )
+
+            request_url = (
+                f"{api_base}/openai/deployments/{deployment_id}"
+                f"/chat/completions?api-version={api_version}"
+            )
+        else:
+            inputs = {"model": endpoint, **inputs}
+            request_url = REQUEST_URL_CHAT
+
+        try:
+            return process_api_requests(
+                [inputs],
+                request_url,
+                api_token=api_token,
+                throw_original_error=True,
+                max_workers=1,
+            )[0]
+        except MlflowException:
+            raise
+        except Exception as e:
+            raise MlflowException(f"Error response from OpenAI:\n {e}")
+
+    def create_endpoint(self, name, config=None):
+        """
+        .. warning::
+
+            This method is not implemented for `OpenAIDeploymentClient`.
+        """
+        raise NotImplementedError
+
+    def update_endpoint(self, endpoint, config=None):
+        """
+        .. warning::
+
+            This method is not implemented for `OpenAIDeploymentClient`.
+        """
+        raise NotImplementedError
+
+    def delete_endpoint(self, endpoint):
+        """
+        .. warning::
+
+            This method is not implemented for `OpenAIDeploymentClient`.
+        """
+        raise NotImplementedError
+
+    def list_endpoints(self):
+        """
+        List the currently available models.
+        """
+
+        _check_openai_key()
+
+        api_config = _get_api_config_without_openai_dep()
+        import requests
+
+        if api_config.api_type in ("azure", "azure_ad", "azuread"):
+            raise NotImplementedError(
+                "List endpoints is not implemented for Azure OpenAI API",
+            )
+        else:
+            api_key = os.environ["OPENAI_API_KEY"]
+            request_header = {"Authorization": f"Bearer {api_key}"}
+
+            response = requests.get(
+                "https://api.openai.com/v1/models",
+                headers=request_header,
+            )
+
+            if response.status_code != 200:
+                rest_utils.augmented_raise_for_status(response)
```

Comment:
```suggestion
            rest_utils.augmented_raise_for_status(response)
```

`augmented_raise_for_status` does nothing if the request is successful.